### PR TITLE
fix: improve service, ingress and cert-manager configs

### PR DESCRIPTION
**Pull request by jgreat**
### Motivation

Cert-Manager documentation wasn't clear on how to "share" an TLS certificate across multiple ingresses. Instead of using ingress annotations, use a common Certificate object and share that object (secrets) on ingresses that use that hostname. Clean up the ingress/service mapping

- ingress: update cert-manager certificates from annotations to shared Certificate objects.
- ingress/service: clean up port mapping and use port "names" instead of hard coding numbers
- ingress: update rate-limit config to automatically use separate buckets based on the consensus node instance number.
- ingress: break out common and specific annotations in values for more DRY config.

- consensus: use "sidecar" pattern for consensus node http-grpc gateway service (like we do in fog). More efficient request routing.
- consensus/service/ingress: sidecar http-grpc gateway service improvements.
- testing/CD: Add test script a the CD deploy to check if services are ready before we continue.

### CODE REVIEW

**Review by samdealy — APPROVED** [review]

File: .internal-ci/helm/consensus-node/templates/client-http-attest-ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ include "consensusNode.fullname" . }}-client-http-attest
+  annotations:
+    {{- if .Values.node.ingress.clientAttest.rateLimits.enabled }}
+    {{- toYaml .Values.node.ingress.clientAttest.rateLimits.annotations | nindent 4 }}
+    haproxy.org/rate-limit-period: {{ include "consensusNode.rateLimitPeriod" . | quote }}
+    {{- end }}
+    {{- if (include "consensusNode.blocklist.enabled" .) }}
```
Comment (jgreat) [inline .internal-ci/helm/consensus-node/templates/client-http-attest-ingress.yaml:12]: eq string "true"

File: .internal-ci/helm/consensus-node/templates/peer-grpc-ingress.yaml [hunk 2]
```diff
@@ -4,20 +4,18 @@ kind: Ingress
 metadata:
   name: {{ include "consensusNode.fullname" . }}-peer-grpc
   annotations:
-    {{- if .Values.global.certManagerClusterIssuer }}
-    cert-manager.io/cluster-issuer: {{ .Values.global.certManagerClusterIssuer }}
-    {{- end }}
-    {{- if eq (include "consensusNode.blocklist.enabled" .) "true" }}
+    {{- if (include "consensusNode.blocklist.enabled" .) }}
```
Comment (jgreat) [inline .internal-ci/helm/consensus-node/templates/peer-grpc-ingress.yaml:7]: eq string "true"

File: .internal-ci/helm/fog-services/templates/fog-ledger-grpc-ingress.yaml [hunk 3]
```diff
@@ -7,18 +7,16 @@ metadata:
     app: fog-ledger
     {{- include "fogServices.labels" . | nindent 4 }}
   annotations:
-    {{- if .Values.global.certManagerClusterIssuer }}
-    cert-manager.io/cluster-issuer: {{ .Values.global.certManagerClusterIssuer }}
-    {{- end }}
-    {{- if eq (include "fogServices.blocklist.enabled" .) "true" }}
+    {{- if (include "fogServices.blocklist.enabled" .) }}
```
Comment (jgreat) [inline .internal-ci/helm/fog-services/templates/fog-ledger-grpc-ingress.yaml:10]: eq string true

File: .internal-ci/helm/fog-services/templates/fog-ledger-http-ingress.yaml [hunk 4]
```diff
@@ -7,18 +7,16 @@ metadata:
     app: fog-ledger
     {{- include "fogServices.labels" . | nindent 4 }}
   annotations:
-    {{- if .Values.global.certManagerClusterIssuer }}
-    cert-manager.io/cluster-issuer: {{ .Values.global.certManagerClusterIssuer }}
-    {{- end }}
-    {{- if eq (include "fogServices.blocklist.enabled" .) "true" }}
+    {{- if (include "fogServices.blocklist.enabled" .) }}
```
Comment (jgreat) [inline .internal-ci/helm/fog-services/templates/fog-ledger-http-ingress.yaml:10]: use eq string "true"

File: .internal-ci/helm/fog-services/templates/fog-report-grpc-ingress.yaml [hunk 5]
```diff
@@ -1,28 +1,27 @@
...
-  name: {{ include "fogServices.fullname" $ }}-fog-report-grpc-{{ . }}
+  name: {{ include "fogServices.fullname" . }}-fog-report-grpc
   labels:
     app: fog-report
-    {{- include "fogServices.labels" $ | nindent 4 }}
+    {{- include "fogServices.labels" . | nindent 4 }}
   annotations:
-    {{- if $.Values.global.certManagerClusterIssuer }}
-    cert-manager.io/cluster-issuer: {{ $.Values.global.certManagerClusterIssuer }}
+    {{- if (include "fogServices.blocklist.enabled" .) }}
```
Comment (jgreat) [inline .internal-ci/helm/fog-services/templates/fog-report-grpc-ingress.yaml:11]: eq string true

File: .internal-ci/helm/fog-services/templates/fog-view-grpc-ingress.yaml [hunk 6]
```diff
@@ -7,18 +7,16 @@ metadata:
     app: fog-view
     {{- include "fogServices.labels" . | nindent 4 }}
   annotations:
-    {{- if .Values.global.certManagerClusterIssuer }}
-    cert-manager.io/cluster-issuer: {{ .Values.global.certManagerClusterIssuer }}
-    {{- end }}
-    {{- if eq (include "fogServices.blocklist.enabled" .) "true" }}
+    {{- if (include "fogServices.blocklist.enabled" .) }}
```
Comment (jgreat) [inline .internal-ci/helm/fog-services/templates/fog-view-grpc-ingress.yaml:10]: eq string true

File: .internal-ci/helm/fog-services/templates/fog-view-http-ingress.yaml [hunk 7]
```diff
@@ -7,18 +7,16 @@ metadata:
     app: fog-view
     {{- include "fogServices.labels" . | nindent 4 }}
   annotations:
-    {{- if .Values.global.certManagerClusterIssuer }}
-    cert-manager.io/cluster-issuer: {{ .Values.global.certManagerClusterIssuer }}
-    {{- end }}
-    {{- if eq (include "fogServices.blocklist.enabled" .) "true" }}
+    {{- if (include "fogServices.blocklist.enabled" .) }}
```
Comment (jgreat) [inline .internal-ci/helm/fog-services/templates/fog-view-http-ingress.yaml:10]: eq string true

File: .internal-ci/test/check-ingress-ready.sh [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        *)
+            echo "${1} unknown option"
+            usage
+            exit 1
+            ;;
+    esac
+done
+
+counter=0
+while ! curl -sSLf -X POST -o /dev/null "${url}"
```
Comment (jgreat) [inline .internal-ci/test/check-ingress-ready.sh:36]: Add timeout

**Review by aweis89 — APPROVED** [review]

**Review by joekottke — APPROVED** [review]
Approved based on walk-through and discussion.  LGTM.

**Pull request merged**